code/Gamemodes/Bowling/Bowling.cs

```diff
@@ -1,15 +1,84 @@
-﻿namespace Sports;
+﻿using Sports.StateSystem;
+
+namespace Sports;
 
 [HammerEntity]
 [Library( "sports_bowling_gamemode" )]
 [EditorModel( "models/Bowling/bpin.vmdl" )]
 [Title( "Bowling Gamemode" )]
 [Category( "Gamemodes" )]
 [Sphere( 128f, 0, 125, 255 )]
-public partial class Bowling : BaseGamemode
+public partial class Bowling : BaseGamemode, IStateMachine<TurnStateMachine>
 {
+	[Net] public TurnStateMachine StateMachine { get; set; }
+
 	public override BasePlayer CreatePawn()
 	{
 		return new BowlingPlayer();
 	}
+	public override void Spawn()
+	{
+		base.Spawn();
+		StateMachine = new()
+		{
+			Gamemode = this,
```

Comment:
Maybe State Machine ctor should just pass in the gamemode instead of an object initializer

Reply:
does that work with entities? im always scared to use the constructor for entities.

Reply:
should be okay as long as you have an empty ctor as well

Reply:
by empty I mean no args